cli/azd/pkg/messaging/topic.go

```diff
@@ -0,0 +1,59 @@
+package messaging
+
+import (
+	"context"
+
+	"golang.org/x/exp/slices"
+)
+
+type Topic struct {
+	Name        string
+	channel     chan *Message
+	subscribers []*Subscription
+}
+
+func NewTopic(name string) *Topic {
+	return &Topic{
+		Name:        name,
+		channel:     make(chan *Message),
+		subscribers: []*Subscription{},
```

Comment:
Topics use go channels under the covers and support multiple subscribers vs a traditional go channel that can only have a single consumer.

Reply:
having multiple subscribers is cool!, but I don't think we would ever have a use case for that within the cli while executing a command.
It means there would be more than one command listening to the messages from an action, which is never a case.

Reply:
> Could you help me better understand it by giving an example of having multiple subscribers?

@hemarina 
An example could be a logging middleware capturing all messages to create a structured log file vs individual command actions only capturing a subset of messages used to format a user friendly UX experience.

In this case the message would be delivered to each subscriber independently and could take their own actions against the message.